### PR TITLE
[Network] Added Cmdlets for Network Virtual Appliance (NVA) and NVA site resources.

### PR DESCRIPTION
**Pull request by 3vilbuff3r**
## Description
This pull request adds cmdlets for a new resource type "Network Virtual Appliance" in Azure.

The link to [powershell design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/594)

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by neelbhavsar97** [comment]
@number213 , I see that some validations are failing due to "online version" field being empty in the help md files. We do not have any online version of the help yet on these commands. Should I just add the expected online help page link in those fields?

**Comment by VeryEarly** [comment]
> @number213 , I see that some validations are failing due to "online version" field being empty in the help md files. We do not have any online version of the help yet on these commands. Should I just add the expected online help page link in those fields?
> 
> Edit: I see that there is an open [issue](https://github.com/Azure/azure-powershell/issues/11766) on this behavior of the platyPS and checks in the CI. Let me know the steps I need to follow to resolve this. :)

The online version should be in certain format, just paste it to help markdown.

**Comment by neelbhavsar97** [comment]
Hi @VeryEarly , can you point me to the required format?. The generated online version is empty.

**Comment by VeryEarly** [comment]
> Hi @VeryEarly , can you point me to the required format?. The generated online version is empty.

should be like this:
https://docs.microsoft.com/en-us/powershell/module/az.network/{cmdlet}

File: src/Network/Network/NetworkVirtualAppliance/GetNetworkVirtualApplianceCommand.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+using Microsoft.Azure.Management.Network.Models;
+using Microsoft.Rest.Azure;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkVirtualAppliance"), OutputType(typeof(PSNetworkVirtualAppliance))]
```
Comment (VeryEarly) [inline src/Network/Network/NetworkVirtualAppliance/GetNetworkVirtualApplianceCommand.cs:29]: please add default parameter

File: src/Network/Network/NetworkVirtualAppliance/NewNetworkVirtualApplianceCommand.cs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Management.Automation.Remoting;
+using System.Text;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkVirtualAppliance", SupportsShouldProcess = true), OutputType(typeof(PSNetworkVirtualAppliance))]
```
Comment (VeryEarly) [inline src/Network/Network/NetworkVirtualAppliance/NewNetworkVirtualApplianceCommand.cs:30]: please add ResourceNameParameterSet as default parameter

File: src/Network/Network/NetworkVirtualAppliance/RemoveNetworkVirtualApplianceCommand.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Network;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkVirtualAppliance", SupportsShouldProcess = true), OutputType(typeof(bool))]
```
Comment (VeryEarly) [inline src/Network/Network/NetworkVirtualAppliance/RemoveNetworkVirtualApplianceCommand.cs:26]: please add ResourceNameParameterSet as default parameter

File: src/Network/Network/NetworkVirtualAppliance/Sites/GetVirtualApplianceSiteCommand.cs [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.",
+            ParameterSetName = ResourceNameParameterSet)]
+        [ResourceGroupCompleter]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
```
Comment (VeryEarly) [inline src/Network/Network/NetworkVirtualAppliance/Sites/GetVirtualApplianceSiteCommand.cs:59]: this should be true

File: src/Network/Network/NetworkVirtualAppliance/Sites/GetVirtualApplianceSiteCommand.cs [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using Microsoft.Rest.Azure;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualApplianceSite", DefaultParameterSetName = ResourceNameParameterSet), OutputType(typeof(PSVirtualApplianceSite))]
```
Comment (VeryEarly) [inline src/Network/Network/NetworkVirtualAppliance/Sites/GetVirtualApplianceSiteCommand.cs:27]: please add ResourceNameParameterSet as default parameter

File: src/Network/Network/NetworkVirtualAppliance/Sites/NewVirtualApplianceSiteCommand.cs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Network;
+using System;
+using System.Collections;
+using System.Management.Automation;
+using MNM = Microsoft.Azure.Management.Network.Models;
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualApplianceSite", SupportsShouldProcess = true), OutputType(typeof(PSVirtualApplianceSite))]
```
Comment (VeryEarly) [inline src/Network/Network/NetworkVirtualAppliance/Sites/NewVirtualApplianceSiteCommand.cs:25]: please add ResourceNameParameterSet as default parameter

File: src/Network/Network/NetworkVirtualAppliance/Sites/Office365PolicyProperties/NewOffice365PolicyPropertiesCommand.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+using Microsoft.Azure.Commands.Network.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "Office365PolicyProperties"), OutputType(typeof(PSOffice365PolicyProperties))]
```
Comment (VeryEarly) [inline src/Network/Network/NetworkVirtualAppliance/Sites/Office365PolicyProperties/NewOffice365PolicyPropertiesCommand.cs:24]: sorry I didn't identify this in design review, but please use singularity for cmdlet name:
New-AzOffice365PolicyProperty

File: src/Network/Network/NetworkVirtualAppliance/Sites/RemoveVirtualApplianceSiteCommand.cs [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Network;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualApplianceSite", SupportsShouldProcess = true), OutputType(typeof(bool))]
```
Comment (VeryEarly) [inline src/Network/Network/NetworkVirtualAppliance/Sites/RemoveVirtualApplianceSiteCommand.cs:26]: please add ResourceNameParameterSet as default parameter

File: src/Network/Network/NetworkVirtualAppliance/SkuProperties/NewVirtualApplianceSkuPropertiesCommand.cs [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+using Microsoft.Azure.Commands.Common.Compute.Version2016_04_preview.Models;
+using Microsoft.Azure.Commands.Network.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualApplianceSkuProperties"), OutputType(typeof(PSVirtualApplianceSkuProperties))]
```
Comment (VeryEarly) [inline src/Network/Network/NetworkVirtualAppliance/SkuProperties/NewVirtualApplianceSkuPropertiesCommand.cs:25]: sorry I didn't identify this in design review, but please use singularity for cmdlet name:
New-AzVirtualApplianceSkiProperties -> New-AzVirtualApplianceSkiProperty

**Comment by neelbhavsar97** [comment]
@VeryEarly , some of the checks are failing due to the commands "New-AzOffice365PolicyProperty" and "New-AzVirtualApplianceSkuProperty". Both commands do NOT create any resources but rather create a local powershell objects to be inserted in a subsequent command. In these cases I believe the options like ShouldProcess, Force, Confirm do not apply. Let me know if my understanding is incorrect and something needs to be changed.

**Comment by VeryEarly** [comment]
> @VeryEarly , some of the checks are failing due to the commands "New-AzOffice365PolicyProperty" and "New-AzVirtualApplianceSkuProperty". Both commands do NOT create any resources but rather create a local powershell objects to be inserted in a subsequent command. In these cases I believe the options like ShouldProcess, Force, Confirm do not apply. Let me know if my understanding is incorrect and something needs to be changed. Also, shall we change the design review issue with the new command names as well?

Hi @neelbhavsar97 ,

Please suppress these two messages into https://github.com/Azure/azure-powershell/blob/master/tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv

Please re-target your PR to [network-june](https://github.com/Azure/azure-powershell/tree/network-june) branch, one simple way is to create a new branch out of network-june and cherry-pick commits into it.

**Comment by neelbhavsar97** [comment]
@VeryEarly , created a new branch and a new [PR](https://github.com/Azure/azure-powershell/pull/12227). Closing this PR.

**Comment by neelbhavsar97** [comment]
@VeryEarly , Some checks are failing with no obvious reason related to the changes I've pushed. Is this expected or I'm missing something here?

**Comment by VeryEarly** [comment]
> @VeryEarly , Some checks are failing with no obvious reason related to the changes I've pushed. Is this expected or I'm missing something here?

Hi @neelbhavsar97 ,

These failures were caused by some recent changes in master branch, I'll take care of it.

**Comment by neelbhavsar97** [comment]
Sure, let me know if you need something. You can go ahead and merge this when convenient. Thanks you.

**Comment by neelbhavsar97** [comment]
Hi @VeryEarly , there is a conflict for this PR. Can you resolve this conflict as it is related to introducing a new version to the Az.Network module. Let me know if you need anything from my side.

**Comment by neelbhavsar97** [comment]
Hi @number213 , please review and merge this pull request as the release date is approaching. Let me know if you need any changes.

**Pull request merged**